crawler.py

```diff
@@ -1,32 +1,43 @@
+import requests
+
 from bs4 import BeautifulSoup
 
-import requests
 
-def parse():
+def crawl():
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 '
+                      '(KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36'
+    }
+
+    games = []
 
-	HEADERS = {
-		'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.129 Safari/537.36'
-	}
+    page_number = 1
+    while True:
+        page_url = f'https://marketplace.xbox.com/en-US/Games/Xbox360Games?Page={page_number}'
 
-	max_page = 43
-	pages = []
+        print(f"Loading page #{page_number}: {page_url}")
+        response = requests.get(page_url,headers=headers)
+        print("Done")
 
-	for x in range(1, max_page + 1):
-		pages.append(requests.get('https://marketplace.xbox.com/en-US/Games/Xbox360Games?Page=' + str(x), headers = HEADERS))
+        soup = BeautifulSoup(response.content, 'html.parser')
 
-	games = []
+        if soup.find_all("div", {"class": "NoGamesFound"}):
+            # if we reached the end, there is a div with class 'NoGamesFound',
+            # so we stop crawling
+            print("No data on page, finishing execution.")
+            break
 
-	for r in pages:
-		soup = BeautifulSoup(r.content, 'html.parser')
-		items = soup.findAll('li', class_ = 'grid-6')
+        for item in soup.findAll('li', class_='grid-6'):
+            rating_str = item.find('div', class_='UserRatingStarStrip').get_text(strip=True)
+            game = dict(
+                title=item.find('h2').get_text(strip=True),
+                rating=float(rating_str.split(' ')[0])
+            )
+            games.append(game)
+            print("Parsed game:", game)
 
-		for item in items:
-			games.append({
-				'title': item.find('h2').get_text(strip = True),
-				'rating': item.find('div', class_ = 'UserRatingStarStrip').get_text(strip = True)
-			})
+        page_number += 1
 
-	for game in games:
-	 	print(game['title'], game['rating'])
 
-parse()
```

Comment:
You should not use such calls, _because_ in case you do `import *` from this file, this function will be executed.

See below a proper way to make function called only when you run file (like `python crawler.py`). ↓